staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/controller_reconcile.go

```diff
@@ -423,8 +427,16 @@ func (c *policyController) latestPolicyData() []policyData {
 		var bindingInfos []bindingInfo
 		for bindingNN := range c.definitionsToBindings[definitionNN] {
 			bindingInfo := c.bindingInfos[bindingNN]
-			if bindingInfo.validator == nil && definitionInfo.configurationError == nil {
-				bindingInfo.validator = c.ValidatorCompiler.Compile(definitionInfo.lastReconciledValue)
+			if bindingInfo.evaluator == nil && definitionInfo.configurationError == nil {
+				hasParam := false
+				if definitionInfo.lastReconciledValue.Spec.ParamKind != nil {
```

Comment:
Can `lastReconciledValue` be nil here? (It's not clear to me, just looking at this code region, if it's safe)

Reply:
I think its safe because it is the same access pattern as the existing code. I just moved it into controller_reconcile.go from validator.go

The previous code has that logic ![here in validator.go](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/validator.go#L121-L124) which is called from ![here in controller_reconcile.go](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/controller_reconcile.go#L426-L428)  without a nil check